src/solver/hydro/management/management.cpp

```diff
@@ -338,6 +326,7 @@ bool HydroManagement::checkMinGeneration(uint numSpace)
         uint z = area.index;
         const auto& ptchro = NumeroChroniquesTireesParPays[numSpace][z];
         auto tsIndex = (uint)ptchro.Hydraulique;
+        auto tsIndexPowerCredits = ptchro.HydrauliquePowerCredits;
```

Comment:
`bool HydroManagement::checkMinGeneration(uint numSpace)` : 
We don't make checks on hourly min generation power, but on min and max hourly generation power.
So I suggest we call **checkMinGeneration** and **checkGenerationPowerConsistency** on the same level : 
```C++
void HydroManagement::operator() (...)
{
    ...
    if (!checkMinGeneration(numSpace))
    {
        throw FatalError("hydro management: invalid minimum generation");
    }
    if (!checkGenerationPowerConsistency(...))
    {
        throw FatalError("hydro management: invalid minimum generation");
    }
    ...
}
```
With a loop on areas inside **checkGenerationPowerConsistency(...)**

Or even better : 
```C++
void HydroManagement::operator() (...)
{
    ...
    if (!checksOnGenerationPowerBounds( /* some args here*/ ))
    {
        throw FatalError("hydro management: invalid minimum generation");
    }
    ...
```
And split **checksOnGenerationPowerBounds(...)** in 2 parts like just above.

Reply:
Done!